cmake/FDBComponents.cmake

```
@@ -75,12 +59,56 @@ else()
   set(WITH_PYTHON OFF)
 endif()
 
+option(BUILD_PYTHON_BINDING "build python binding" ON)
+if(NOT BUILD_PYTHON_BINDING OR NOT WITH_PYTHON)
+  set(WITH_PYTHON_BINDING OFF)
+else()
+  if(WITH_PYTHON)
+    set(WITH_PYTHON_BINDING ON)
+  else()
+    #message(FATAL_ERROR "Could not found a suitable python interpreter")
+    set(WITH_PYTHON_BINDING OFF)
+  endif()
+endif()
+
+################################################################################
+# C Bindings
+################################################################################
+
+option(BUILD_C_BINDING "build C binding" ON)
+if(BUILD_C_BINDING AND WITH_PYTHON)
+  set(WITH_C_BINDING ON)
+else()
+  set(WITH_C_BINDING OFF)
+endif()
+
+################################################################################
+# Java Bindings
+################################################################################
+
+option(BUILD_JAVA_BINDING "build java binding" ON)
+if(NOT BUILD_JAVA_BINDING OR NOT WITH_C_BINDING)
+  set(WITH_JAVA_BINDING OFF)
+else()
+  set(WITH_JAVA_BINDING OFF)
+  find_package(JNI 1.8)
+  find_package(Java 1.8 COMPONENTS Development)
+  # leave FreeBSD JVM compat for later
+  if(JNI_FOUND AND Java_FOUND AND Java_Development_FOUND AND NOT (CMAKE_SYSTEM_NAME STREQUAL "FreeBSD") AND WITH_C_BINDING)
+    set(WITH_JAVA_BINDING ON)
+    include(UseJava)
+    enable_language(Java)
+  else()
+    set(WITH_JAVA_BINDING OFF)
+  endif()
+endif()
+
 ################################################################################
 # Pip
 ################################################################################
 
-find_package(Python3 COMPONENTS Interpreter)
-if (Python3_Interpreter_FOUND)
+option(BUILD_DOCUMENTATION "build documentation" ON)
+if (BUILD_DOCUMENTATION AND WITH_PYTHON)
```

we use `${Python3_EXECUTABLE}` in `documentation/CMakeLists.txt`. Removing the call to `find_package(Python3)` will make this then fail (so the documentation doesn't build anymore.

The easiest solution would be to reintroduce that call, though that is not pretty. Alternatively you could do the following:

1. Replace `${Python3_EXECUTABLE}`
2. Only set `WITH_PYTHON` to `ON` if `Python_VERSION_MAJOR` is 3 or:

```suggestion
option(BUILD_DOCUMENTATION "build documentation" ON)
if (BUILD_DOCUMENTATION AND WITH_PYTHON AND Python_VERSION_MAJOR EQUAL 3)
```